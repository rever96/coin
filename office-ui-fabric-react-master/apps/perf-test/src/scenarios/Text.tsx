import * as React from 'react';
import { Text } from 'office-ui-fabric-react';

const scenario = <Text>I am text, hear me roar.</Text>;

export default scenario;
