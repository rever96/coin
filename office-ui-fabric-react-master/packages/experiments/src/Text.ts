/**
 * @deprecated Import from office-ui-fabric-react package instead.
 */
export { Text } from 'office-ui-fabric-react/lib/Text';
