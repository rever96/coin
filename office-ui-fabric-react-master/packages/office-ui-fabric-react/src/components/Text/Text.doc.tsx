import * as React from 'react';
import { TextRampExample } from './examples/Text.Ramp.Example';
import { TextWrapExample } from './examples/Text.Wrap.Example';
import { TextBlockExample } from './examples/Text.Block.Example';

import { IDocPageProps } from '../../common/DocPage.types';

const TextRampExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Text/examples/Text.Ramp.Example.tsx') as string;

const TextWrapExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Text/examples/Text.Wrap.Example.tsx') as string;

const TextBlockExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/Text/examples/Text.Block.Example.tsx') as string;

export const TextPageProps: IDocPageProps = {
  title: 'Text',
  componentName: 'Text',
  componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/Text',
  examples: [
    {
      title: 'Text Ramp Example',
      code: TextRampExampleCode,
      view: <TextRampExample />
    },
    {
      title: 'Text Wrap Example',
      code: TextWrapExampleCode,
      view: <TextWrapExample />
    },
    {
      title: 'Text Block Example',
      code: TextBlockExampleCode,
      view: <TextBlockExample />
    }
  ],
  overview: require<string>('!raw-loader!office-ui-fabric-react/src/components/Text/docs/TextOverview.md'),
  bestPractices: '',
  dos: require<string>('!raw-loader!office-ui-fabric-react/src/components/Text/docs/TextDos.md'),
  donts: require<string>('!raw-loader!office-ui-fabric-react/src/components/Text/docs/TextDonts.md'),
  isHeaderVisible: true,
  isFeedbackVisible: true
};
