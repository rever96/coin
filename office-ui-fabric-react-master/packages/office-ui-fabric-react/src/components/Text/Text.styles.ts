import { ITextComponent, ITextStyles, ITextStylesReturnType, ITextProps } from './Text.types';

import { ITheme } from '../../Styling';

export const TextStyles: ITextComponent['styles'] = (props: ITextProps, theme: ITheme): ITextStylesReturnType => {
  const { as, className, block, nowrap, variant } = props;
  const { fonts } = theme;
  const variantObject = fonts[variant || 'medium'];

  return {
    root: [
      theme.fonts.medium,
      {
        display: block ? (as === 'td' ? 'table-cell' : 'block') : 'inline',
        fontFamily: (variantObject && variantObject.fontFamily) || 'inherit',
        fontSize: (variantObject && variantObject.fontSize) || 'inherit',
        fontWeight: (variantObject && variantObject.fontWeight) || 'inherit',
        color: (variantObject && variantObject.color) || 'inherit',
        mozOsxFontSmoothing: variantObject && variantObject.MozOsxFontSmoothing,
        webkitFontSmoothing: variantObject && variantObject.WebkitFontSmoothing
      },
      nowrap && {
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis'
      },
      className
    ]
  } as ITextStyles;
};
