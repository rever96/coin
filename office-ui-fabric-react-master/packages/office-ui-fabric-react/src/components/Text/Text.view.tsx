/** @jsx withSlots */
import { withSlots, getSlots } from '../../Foundation';
import { getNativeProps, htmlElementProperties } from '../../Utilities';
import { ITextComponent, ITextProps, ITextSlots } from './Text.types';

export const TextView: ITextComponent['view'] = props => {
  if (!props.children) {
    return null;
  }

  const { block, className, as: RootType = 'span', variant, nowrap, ...rest } = props;

  const Slots = getSlots<ITextProps, ITextSlots>(props, {
    root: RootType
  });

  return <Slots.root {...getNativeProps(rest, htmlElementProperties)} />;
};
