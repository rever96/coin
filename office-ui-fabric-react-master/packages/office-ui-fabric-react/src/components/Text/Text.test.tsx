import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { Text } from './Text';

describe('Text', () => {
  it('renders default Text correctly', () => {
    const component = renderer.create(<Text>I'm default text</Text>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
