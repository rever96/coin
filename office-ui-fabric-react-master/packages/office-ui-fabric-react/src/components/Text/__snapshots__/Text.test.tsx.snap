// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders default Text correctly 1`] = `
<span
  className=

      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        color: inherit;
        display: inline;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
        moz-osx-font-smoothing: grayscale;
        webkit-font-smoothing: antialiased;
      }
>
  I'm default text
</span>
`;
